{
  try {
    ClassFieldAccessor intAccessor=cache.getAccessor(TestClass.class,"intAttr",getClass().getClassLoader());
    ClassFieldAccessor strAccessor=cache.getAccessor(TestClass.class,"strAttr",getClass().getClassLoader());
    String testString1="TestAttr1";
    String testString2="TestAttr2";
    TestClass instance=new TestClass();
    instance.setIntAttr(10);
    instance.setStrAttr(testString1);
    Assert.assertEquals("Error reading int attr",10,((Integer)intAccessor.getValue(instance)).intValue());
    Assert.assertEquals("Error reading int attr",10,intAccessor.getIntValue(instance));
    Assert.assertEquals("Error reading String attr",testString1,strAccessor.getValue(instance));
    intAccessor.setValue(instance,new Integer(50));
    strAccessor.setValue(instance,testString2);
    Assert.assertEquals("Error setting int attr",50,instance.getIntAttr());
    Assert.assertEquals("Error setting String attr",testString2,instance.getStrAttr());
    intAccessor.setIntValue(instance,40);
    Assert.assertEquals("Error reading int attr",40,intAccessor.getIntValue(instance));
    Assert.assertEquals("Error reading String attr",testString2,strAccessor.getValue(instance));
  }
 catch (  Exception e) {
    e.printStackTrace();
    Assert.fail("No exception is supposed to be generated when creating field accessor: " + e);
  }
}
